// copyright and license inspection - no issues 4/13/22

syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.bubblesnet.bubblesgrpc";
option java_outer_classname = "BubblesGRPCProto";

package bubblesgrpc;

// The service definition.
service SensorStoreAndForward {
  //
  rpc StoreAndForward (SensorRequest) returns (SensorReply) {}
  rpc GetState (GetStateRequest) returns (GetStateReply) {}
  rpc GetRecordList (GetRecordListRequest) returns (GetRecordListReply) {}
}

// The request message
message SensorRequest {
  int32 sequence = 1;
  string type_id = 2;
  string data = 3;
}

// The response message containing the greetings
message SensorReply {
  int32 sequence = 1;
  string type_id = 2;
  string result = 3;
  string message = 4;
}

// The request message
message GetStateRequest {
  int32 sequence = 1;
  string type_id = 2;
  string data = 3;
}

// The response message containing the greetings
message GetStateReply {
  int32 sequence = 1;
  string type_id = 2;
  string result = 3;
  float tempF = 4;
  float humidity = 5;
  float externalTempF = 6;
  float waterTempF = 7;
}

// The request message
message GetRecordListRequest {
  int32 sequence = 1;
  string type_id = 2;
  string data = 3;
}

// The response message containing the greetings - query defined in request.data, json results in reply.data
message GetRecordListReply {
  int32 sequence = 1;
  string type_id = 2;
  string result = 3;
  string data = 4;
}
